### ISSUE TITLE
Display page tab text labels even on mobile view, whenever there are 3 or fewer tabs

### ISSUE THREAD

**Issue by amandacilek** [medic/cht-core#5804]
It is hard for users to know which page of the app they are on when their only clue is an icon. Icons can sometimes be ambiguous, and trying to remember their meaning increases cognitive load. We already display page text labels on desktop views, but on small mobile devices the labels disappear (and the vast majority of our users access the app via phones).

We should tweak our design so that we can display page tab text labels even on small mobile views, whenever there are 3 or less tabs. This would add text labels for most CHW views, since most CHWs only see 3 tabs. Users with 4 or more tabs accessing via a small mobile device will still have too many tabs to display text labels, so they will continue to see icons only. 

To accomplish this with our current fontawesome nav icons, slightly decrease the size of the icons, move them up a few pixels, and slightly increase the height of the black mobile header bar to 60px. Text, if displayed, is 12px.
![Header with labels](https://user-images.githubusercontent.com/26235957/61237083-7bbe3100-a6ee-11e9-9092-27d02be835e4.png)

To accomplish this with our new icons, as implemented in #5535, keep the icons at their standard 24 x 24 size, move them up a few pixels, and slightly increase the height of the black mobile header bar to 60px. Text, if displayed, is 12px.
![Header with labels Copy](https://user-images.githubusercontent.com/26235957/61237086-7fea4e80-a6ee-11e9-8f7b-dc28335ea74e.png)

**Comment by MaxDiz** [comment]
@garethbowen this is a good add for v3.8 to reduce any icon redesign confusion

**Comment by garethbowen** [comment]
Yeah let's do it at the same time as replacing fontawesome.

**Comment by SCdF** [comment]
Ready for AT in `5804-tab-labels-on-mobile`

AT:
 - Read the rules noted above around label visibility and make sure it's followed in all cases

**Comment by ngaruko** [comment]
@peter-ochieng - It seems that the there has been a regression here. The tabs still shows  for a user (CHW in this case) even when permissions are off .
 ![image](https://user-images.githubusercontent.com/6979995/74884211-81dc6480-53d7-11ea-9f5b-f9105e2e5b0e.png)
Then when one clicks on the tab, there is an error as in the screenshot. 

![image](https://user-images.githubusercontent.com/6979995/74884160-67a28680-53d7-11ea-93c0-59fb793b1ddd.png)

**Comment by peter-ochieng** [comment]
@SCdF Could you please re-review before I send it back to AT?

**Comment by SCdF** [comment]
This is once again ready for AT

**Comment by brad1905** [comment]
@peter-ochieng Working as expected, 3 tabs on mobile shows text labels
<img width="349" alt="Screen Shot 2020-03-30 at 3 41 55 PM" src="https://user-images.githubusercontent.com/62311678/78076640-a96e0780-735b-11ea-81ba-6bf364ea70c1.png">

**Comment by SCdF** [comment]
@brad1905 that layout looks a little squashed, with the title texts being cut off at the bottom. Did you see this consistently or do you think this is an outlier worth ignoring?

**Comment by ngaruko** [comment]
@SCdF We have noticed that and discussed it with @brad1905 - Depending on the form factor, the text is quite squashed . I guess there needs to be a bit of a consistent padding around that.

**Comment by SCdF** [comment]
@peter-ochieng back to you to take a look at and see if you can reproduce and fix this

@ngaruko / @brad1905 can you provide more information on in what situations you've seen this: you say "depending on form factor", please list those form factors to help reproduction.

**Comment by brad1905** [comment]
<img width="385" alt="5804 Galaxy view" src="https://user-images.githubusercontent.com/62311678/79167619-ba2b6e00-7d9c-11ea-8529-05560be2d172.png">

**Comment by garethbowen** [comment]
@peter-ochieng It also looks like the bubbles are slightly separated from the icon. Can you check the red bubbles are in the same place as master?

**Comment by peter-ochieng** [comment]
@brad1905 Can you have another look,I've made updates for the squashed text and bubbles spacing that @garethbowen mentioned.
<img width="349" alt="Bubbles and Label spacing" src="https://user-images.githubusercontent.com/864229/80098071-f2893400-8574-11ea-9aac-a52da9008d92.png">

Thanks

**Comment by SCdF** [comment]
@peter-ochieng your changes mean that you see bubbles even if there are zero:

Master:
![image](https://user-images.githubusercontent.com/583851/80590177-5eefb180-8a13-11ea-8e64-767ddf2d5e52.png)

Your branch:
![image](https://user-images.githubusercontent.com/583851/80590509-e50bf800-8a13-11ea-8d0f-f49c7b169577.png)

Also, the way you've moved stuff around is now cutting off the bottom of the filter layer:

![image](https://user-images.githubusercontent.com/583851/80591239-3cf72e80-8a15-11ea-9bb0-684ec2514cec.png)

This only happens if you use the very smallest responsive layout. In my case: the default Galaxy S5 360x640 profile that comes with Chrome. Using a larger one looks better, but it also doesn't trigger your change at all (ie in larger layouts we already say tab labels)

**Comment by peter-ochieng** [comment]
@SCdF Can you please re-review & AT.

**Comment by SCdF** [comment]
@peter-ochieng this still isn't working, I think it's time to hand this off to someone else or pair with them to go through this, perhaps @garethbowen can lend a hand here? 

Here are the problems that I see:

This was in your previous attempt but I didn't notice at the time: you've dropped the highlight on the current active tab, in all resolutions (your branch on left, master on right):

![image](https://user-images.githubusercontent.com/583851/81045905-1fabdf80-8eaf-11ea-9e99-8dff4bdf1408.png)

You're also holding the selection colour continuously after clicking. This is hard to explain, but in the above screenshot the Tasks button is the blue of its tab (if I selected the people tab it would be orange). This doesn't happen on first refresh, only on clicking, and this colouring exists in master but gets overridden almost instantly by the highlight taking effect.

While the text now displays (your branch on left, master on right):

![image](https://user-images.githubusercontent.com/583851/81046035-66013e80-8eaf-11ea-9592-8be6c411d7d8.png)

And the filter bar also appears correct:

![image](https://user-images.githubusercontent.com/583851/81046514-7960d980-8eb0-11ea-837d-f5005332a668.png)

The text is awkward and squashed. See how different it is on the slightly higher normal resolution (your branch <400px on left, your branch >400px on right):

![image](https://user-images.githubusercontent.com/583851/81046154-a19c0880-8eaf-11ea-8601-5b60c8b9c777.png)

**Comment by peter-ochieng** [comment]
@SCdF Could you have a look?the raised issues were fixed and I worked with @garethbowen to fix the styling issues as well.
Thanks

**Comment by SCdF** [comment]
@peter-ochieng I think we've cracked it!

AFAICT this works on Chromium and Firefox on desktop, as well as the Y4 and Moto G4.

Side note: I've noticed that the tecno y4 has a rendering issue where it shows a small black line, but it's also present on master, so :shrug:. Putting it here in case @garethbowen you think it's worth raising something over (imo it's not: esp since it's hard to debug that phone now and we want to move on from it).

<details>
<summary>Tecno Y4 horizontal line bug</summary>

![Screenshot_2020-05-11-12-17-02](https://user-images.githubusercontent.com/583851/81556543-6bb0c580-9382-11ea-83a5-7687fcedb7e6.png)

</details>
